### PR TITLE
PYTHON-3111 Rename "Versioned API" to "Stable API" in documentation

### PR DESCRIPTION
**Pull request by juliusgeo**

### CODE REVIEW

**Review by behackett — CHANGES_REQUESTED** [review]

File: pymongo/pool.py [hunk 1]
```diff
@@ -590,7 +590,7 @@ def unpin(self):
 
     def hello_cmd(self):
         # Handshake spec requires us to use OP_MSG+hello command for the
-        # initial handshake in load balanced or versioned api mode.
+        # initial handshake in load balanced or versioned API mode.
```
[the remark this answers is not in the document]
Comment (juliusgeo) [reply pymongo/pool.py]: Done.

**Review by behackett — APPROVED** [review]
lgtm

**Review by blink1073 — CHANGES_REQUESTED** [review]
"Versioned API" is also in `.evergreen/config.yml`, `test/__init__.py`, and `test/test_examples.py`

File: pymongo/pool.py [hunk 2]
```diff
@@ -590,7 +590,7 @@ def unpin(self):
 
     def hello_cmd(self):
         # Handshake spec requires us to use OP_MSG+hello command for the
-        # initial handshake in load balanced or versioned api mode.
```
Comment (juliusgeo) [inline pymongo/pool.py:593]: @behackett should I be changing comments? They are not user facing.

Comment (behackett) [reply pymongo/pool.py:593]: Might as well to avoid confusion from future maintainers.

Comment (juliusgeo) [reply pymongo/pool.py:593]: Sounds good.

**Review by blink1073 — APPROVED** [review]
LGTM!

**Review by blink1073 — APPROVED** [review]

File: .evergreen/resync-specs.sh [hunk 3]
```diff
@@ -134,7 +134,7 @@ do
     uri|uri*options)
       cpjson uri-options/tests uri_options
       ;;
-    versioned-api)
+    stable-api)
       cpjson versioned-api/tests versioned-api
```
Comment (ShaneHarvey) [inline .evergreen/resync-specs.sh:138]: Should we wait until the spec changes get merged so that we can update this line as well?

Comment (juliusgeo) [reply .evergreen/resync-specs.sh:138]: Sounds good to me.

Comment (ShaneHarvey) [reply .evergreen/resync-specs.sh:138]: Let's just remove this change and merge.

**Review by ShaneHarvey — APPROVED** [review]
LGTM

**Pull request merged**